src/services/asset-discovery-service.ts

```diff
@@ -150,12 +150,24 @@ export default class AssetDiscoveryService extends PercyClientService {
     })
 
     const maybeResourcePromises: Array<Promise<any>> = []
-    page.on('response', async (response) => {
-      // Parallelize the work in processResponse as much as possible, but make sure to
-      // wait for it to complete before returning from the asset discovery phase.
-      const promise = this.responseService.processResponse(rootResourceUrl, response, width)
-      promise.catch(logError)
-      maybeResourcePromises.push(promise)
+    // Listen on 'requestfinished', which tells us a request completed successfully.
+    // We could also listen on 'response', but then we'd have to check if it was successful.
+    page.on('requestfinished', async (request) => {
+      const response = request.response()
+      if (response != null) {
```

Comment:
Curious, does this compile out to `!==` and we can use `!=` because it's TypeScript?

Reply:
I think just changing this to:

```typescript
if (response) {
```

is probably the simplest approach. Took a quick poke through https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html

Reply:
Yeah, not sure why I wrote it that way, I blame too much language-switching. Fixed now (and was able to remove the non-null coercion after fixing the null check).